electron-app/src/server/websites/e621/e621.service.ts

```diff
@@ -184,4 +275,116 @@ export class e621 extends Website {
 
     return { problems, warnings };
   }
+
+  private validateTag(tagMeta: e621Tag, problems: string[], warnings: string[]) {
+    if (tagMeta.category === e621TagCategory.Invalid) {
+      problems.push(
+        `Tag ${tagMeta.name} is invalid. See https://e621.net/wiki_pages/show_or_new?title=${tagMeta.name}`,
+      );
+    }
+
+    if (tagMeta.post_count < 2) {
+      warnings.push(
+        `Tag ${tagMeta.name} has ${tagMeta.post_count} post(s). Tag may be invalid or low use`,
+      );
+    }
+  }
+
+  private async getUserFeedback(username: string) {
+    return this.getMetdata<e621UserFeedbacksEmpty | e621UserFeedbacks>(
+      `user_feedbacks.json?search[user_name]=${username}`,
+    );
+  }
+
+  private async getTagMetadata(formattedTags: string[]) {
+    return this.getMetdata<e621TagsEmpty | e621Tags>(
+      `tags.json?search[name]=${formattedTags.join(',')}`,
+    );
+  }
+
+  private async getMetdata<T>(url: string) {
+    // TODO Maybe replace with fetch for better performance
+    // TODO Maybe use caching?
```

Comment:
caching would probably be the better option to improve performance, and reduce http calls to e621 in general.